cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -66,7 +66,7 @@ void reader::impl::decode_page_data(bool uses_custom_row_bounds, size_t skip_row
     // TODO: we could probably dummy up size stats for FLBA data since we know the width
     auto const has_flba =
       std::any_of(pass.chunks.begin(), pass.chunks.end(), [](auto const& chunk) {
-        return (chunk.data_type & 7) == FIXED_LEN_BYTE_ARRAY && chunk.converted_type != DECIMAL;
+        return chunk.physical_type == FIXED_LEN_BYTE_ARRAY && !chunk.logical_type.has_value();
```

Comment:
I don't get how these conditions are equivalent (I'm sure they are, I just don't see why)

Reply:
Nice catch, they're not. I was originally intending to implement some other logical types, so rather than checking for decimal, the thought was if there is _any_ logical type set for FLBA, then don't consider it a string. But that's too simplistic, and doesn't allow for logical types we don't understand from just being turned into uninterpreted byte arrays. I think I'll changes this to call something like `is_supported_logical_type()`.

Reply:
fixed